python/lsst/faro/utils/stellarLocus.py

```diff
@@ -0,0 +1,223 @@
+# This file is part of faro.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+import scipy.stats as scipyStats
+import scipy.odr as scipyODR
+from lsst.pipe.base import Struct
+
+__all__ = (
+    "stellarLocusFit",
+    "perpDistance",
+)
+
+
+def stellarLocusFit(xs, ys, paramDict):
```

Comment:
This shouldn't be here and in analysis_drp. I think we discussed moving stuff to a new package analysis that faro, analysis_drp and future analysis packages can import from, for example if analysis_ap becomes a thing.

Reply:
Or you can just import it from analysis_drp.

Reply:
I rewrote/adapted your fitting from analysis_drp as a standalone stellar locus fitting function. As it is written in analysis_drp, the plotting and fitting are intertwined, so it cannot simply be imported on its own and used for faro.

My recollection of our past conversations was that we agreed the fitting/calculation routine should reside somewhere that it can be called from both analysis_drp and faro. We had further agreed that since faro is already in lsst_distrib, the utility functions to do calculations should be placed there. I don't remember discussion of creating a new package, but we could discuss that for the future if that seems useful.